src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,10 +22,58 @@ Retrieve a project's configuration.
 
 **Feature Object**
 
-| Field | Type   | Description                        |
-| ----- | ------ | ---------------------------------- |
-| key   | string | The name used to lookup a feature. |
-| value | any    | A JSON value.                      |
+| Field    | Type           | Description                                                                     |
+| -------- | -------------- | ------------------------------------------------------------------------------- |
+| key      | string         | The name used to lookup a feature.                                              |
+| value    | any            | A JSON value.                                                                   |
+| variants | array[Variant] | Features can have any number of possible variants which alter the return value. |
+
+Features may specify zero or more variants which may alter the evaluation result.
+
+**Variant Object**
+
+| Field   | Type        | Description                                                                               |
+| ------- | ----------- | ----------------------------------------------------------------------------------------- |
+| rollout | Rollout     | An object which contains instructions for rolling out a variant.                          |
+| rules   | array[Rule] | An array of rules which all must evaluate to true in order to return the variant's value. |
+| value   | any         | The value of the feature if the variant is successfully evaluated.                        |
+
+When evaluating a set of rules and varaints each rule is a logical `AND` while each variant is a logical `OR`. Failure to evaluate one variant does not prevent another variant from successfully evaluating. Failure to evaluate one rule does invalidate the remaining rules and the evaluation proceedure can eagerly terminate.
+
+**Rollout Object**
+
+| Field      | Type   | Description                                                 |
+| ---------- | ------ | ----------------------------------------------------------- |
+| percentage | number | The percentage of requests which should evaluate to true.   |
+| seed       | string | A SHA1 hash used to seed a random number generator.         |
+| target     | string | The name of a property contained within the context object. |
```

Comment:
I'm thinking it might make more sense to put `seed` and `target` higher up in the schema. If each variant had it's own pair then it might get weird math wise. Like, if you have no conditions on any variant... and all percentages add up to 100% then you still might have users who get none of the variants because of hashing.

Reply:
Rollout is not a distribution mechanism so there's no concern with the percentages adding up to 100%.  Percentages can sum up to any values greater than 0.

Reply:
So order is 1. rollout first, 2. evaluate conditions... if both true return `"value"` in the conditional, else return the top-level `"value"`?

Or another way, it's something like "percent of the time we should dive in and evaluate if the conditions are true or not"
sounds good.

Reply:
Else check the next variant; if no variants match return the top level value.

> "percent of the time we should dive in and evaluate if the conditions are true or not"

Exactly.